connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -303,14 +303,15 @@ message Expression {
 message CommonInlineUserDefinedFunction {
   // (Required) Name of the user-defined function.
   string function_name = 1;
-  // (Required) Indicate if the user-defined function is deterministic.
+  // (Optional) Indicate if the user-defined function is deterministic.
   bool deterministic = 2;
```

Comment:
JavaUDF has no `deterministic` field but the server doesn't have logic that relies on that field. So only the comment is changed.